ebu_tt_live/bindings/_ebuttdt.py

```diff
@@ -319,11 +319,12 @@ def as_timedelta(cls, instance):
         :param instance:
         :return:
         """
-        hours, minutes, seconds, milliseconds = map(
-            lambda x: cls._int_or_none(x),
-            cls._groups_regex.match(instance).groups()
-        )
-        return timedelta(hours=hours, minutes=minutes, seconds=seconds, milliseconds=milliseconds)
+        hours_str, minutes_str, seconds_str, seconds_fraction_str = [x for x in cls._groups_regex.match(instance).groups()]
+        milliseconds = seconds_fraction_str and cls._int_or_none('{:0<3}'.format(seconds_fraction_str)[:3]) or 0
```

Comment:
Decimal places beyond the third one are truncated here, so e.g. a timecode of `12:34:56.399999` is interpreted as `12:34:56.399`. It could make sense to instead refer to the fourth decimal place for proper rounding (leading to `12:34:56.400` here), in order to fix timestamps of imprecise handling upstream. So maybe something like `int(round(float('0.' + seconds_fraction_str), 3) * 1000)`.

Reply:
Interesting approach, I'll try that.

Reply:
Seems to work. We don't have a requirement for the 3 decimal places constraint so I just use a float in 50ce306.